docs/preview-builds.md

```diff
@@ -0,0 +1,33 @@
+# Testing monorepo changes against other projects
+
+Let's say that you're working on a feature in one of our products and this feature relies on changes that you need to make to one or more packages within this repo. You make those changes here, but now you want to make sure that they fulfill the need you have in your product and don't break anything. How do you integrate these changes into your product?
```

Comment:
I actually had to do a reread to get this one. Let's see if we can make it more straightforward. I also think it can work better within a context (maybe it can be a section under `common_tasks.md` rather than an independent file?)

(*Initial attempt, I'm not proposing it*):

```suggestion
Commonly, you will want to use controllers in external projects, including changes you are working on. There are different ways to do this. When working locally, you may be using [npm](https://docs.npmjs.com/cli/v9/commands/npm-link) or [yarn link](https://classic.yarnpkg.com/lang/en/docs/cli/link/), or changing the `dependencies` in `package.json` of your downstream package to something like `"@metamask/foo-controller": "file:///home/user/src/metamask/packages/foo-controller"`.

This will usually not work easily in instrumented CI builds and other more restricted environments.
```
...Then I started writing about hosting a private registry using [Verdaccio](https://verdaccio.org/) as an alternative.

Do you think it might make more sense to mention preview builds as one option among several (in which the instructions below can still be its own subsection)? I'd be happy to take a stab at part of it if it sounds good to you @mcmire - I think it shouldn't be too hard to get an unpermissioned option going as well.

Reply:
And to you point in the PR comment, the contents of this file is what I would say might belong better in a blog post or article.

Reply:
I found this a bit challenging to read as well.

Just to clarify though, the solution we're describing here _is_ using a private registry. We're using the GitHub Packages registry. This "preview build" system is just a convenient way of publishing arbitrary commits.

Rather than manually publishing, a user can instead type `@metamaskbot preview-build`, and it will build and publish everything for you.

Regarding alternatives, publishing is the only strategy that we've found to work with CI. For local development there are alternatives, though it's not easy to explain all of the caveats to getting those working either.

Reply:
This article might be improved by summarizing the problem scenario more concisely, then proceeding directly to the solution rather than enumerating alternatives first. We can leave alternative strategies for the end.

e.g.
```suggestion
Commonly, you will want to test your changes both locally and in CI before they are merged or published. Our use of the `workspace:` protocol prevents most common strategies from working because it doesn't get resolved until the package is published. We have setup scripts to publish "preview" builds to a private registry to make it easier to test changes while they are still under review.
```

Reply:
I do agree with the reasoning of your last proposal @Gudahtt . Just two things:

1. Can this be communicated without using "we"/"us"?
2. "We have setup scripts" could be read as referring to `yarn setup`, resulting in confusion. Making one of the words into a link to the relevant script would make things a lot more clear.